src/posh-git.psm1

```diff
@@ -1,5 +1,10 @@
 param([bool]$ForcePoshGitPrompt, [bool]$UseLegacyTabExpansion)
 
+if (Test-Path Env:\POSHGIT_ENABLE_STRICTMODE) {
+    # Set strict mode to latest to help catch scripting errors in the module. This is done by the Pester tests.
+    Set-StrictMode -Version Latest
+}
```

Comment:
I assume `Set-StrictMode` is best done here rather than in `test/Shared.ps1` to scope strictness to the module, rather than all tests too?

Hi, hello, yes it's me I still exist. Barely. 👋

Reply:
Yeah, I found when I set it in Shared.ps1 that A) it complained about some of our "lose" scripting in the tests and B) it wasn't finding issues in the module.  That goes counter to the claim in #758 but it makes sense that enabling set-strictmode in the module just for testing is the way to go - IMO.   

BTW I don't think we want to leave strict mode on for ship because it will gen errors with user injected code like this:
```
$global:GitPromptSettings.DefaultPromptSuffix.Text = ' $((Get-History -Count 1).id + 1)$(">" * ($nestedPromptLevel + 1)) '
```
For example, when you first start PowerShell there is no history so `Get-HIstory` returns `$null` and strict-mode doesn't like the null de-ref to get the `id`. But for my prompt string it works out - `$null + 1 ` is `1`.